conjure-go-client/httpclient/body_handler.go

```diff
@@ -116,6 +117,14 @@ func (b *bodyMiddleware) readResponse(resp *http.Response, respErr error) error
 		return nil
 	}
 
+	// Verify the response status. If the response is redirect, no need to decode the body
+	// the client will retry on the new location in the header
+	// TODO: could remove this once #81 is resolved
+	if resp.StatusCode == internal.StatusCodeRetryOther ||
```

Comment:
I think doing this may cause us to leak resp bodies? I think we need to read the body out fully to kill the stream but I may be wrong (wasn't a part of that fix) -- tagging @bmoylan

Reply:
but in this case we will not start a stream to read the body at all from my understanding? I might be completely wrong though. the decoder below will actually read the full body to the buffer and do things.